lightly/utils/optim.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any, Dict, Iterable, Optional
+
+from torch.optim import Optimizer
+
+
+def update_param_groups(
```

Comment:
I think this interface works. It's also similar to how the optimizer is initialized so that's good. 

Did you also consider something like this?
```python
def update_param_groups(
    optimizer: Optimizer,
    update: Optional[Dict[str, Any]] = None,
    filter_fn: Callable[[Dict[str, Any]], bool] = lamba x: x, # lazy default value, we can do better
) -> None:
    for param_group in param_groups:
        if not filter_fn(param_group):
            # Do update here.
            pass
```


Usage could look like this:
```python
update_param_groups(
    optimizer=optimizer,
    update={"weight_decay": 0.5},
    filter_fn=lambda x: "weight_decay" not in x or x["weight_decay"] == 0.0
)

Reply:
I didn't consider it, but I am also not sure if the lambda approach is preferable. I think it will be difficult to specify different updates. For example, how would we update `{"name": "head1", "weight_decay": 0.1}` and `{"name": "head2", "weight_decay": 0.2},`?

Reply:
You would use two calls.

Reply:
No you can do it in a single call:
```
update_param_groups(
    optimizer,
    updates=[
        {"name": "head1", "weight_decay": 0.1},
        {"name": "head2", "weight_decay": 0.2},
    ]
)
```